cargo-shuttle/src/lib.rs

```diff
@@ -954,35 +947,49 @@ impl Shuttle {
         // TODO: Make get_service_summary endpoint wait for a bit and see if it entered Running/Crashed state.
         tokio::time::sleep(std::time::Duration::from_millis(500)).await;
 
-        let service = client.get_service(self.ctx.project_name()).await?;
+        let deployment = client
+            .get_deployment_details(self.ctx.project_name(), &deployment.id)
+            .await?;
 
         // A deployment will only exist if there is currently one in the running state
-        if let Some(ref new_deployment) = service.deployment {
+        if deployment.state == shuttle_common::deployment::State::Running {
+            let service = client.get_service(self.ctx.project_name()).await?;
+
             let resources = client
                 .get_service_resources(self.ctx.project_name())
                 .await?;
             let resources = get_resources_table(&resources, self.ctx.project_name().as_str());
 
             println!("{resources}{service}");
 
-            Ok(match new_deployment.state {
-                shuttle_common::deployment::State::Crashed => CommandOutcome::DeploymentFailure,
-                _ => CommandOutcome::Ok,
-            })
+            Ok(CommandOutcome::Ok)
         } else {
             println!("{}", "Deployment has not entered the running state".red());
             println!();
-            match last_state {
-                Some(shuttle_common::deployment::State::Stopped) => {
+
+            let deployment = client
+                .get_deployment_details(self.ctx.project_name(), &deployment.id)
+                .await?;
```

Comment:
Why should we query for it again?

Reply:
Woops, that's a leftover from testing, thanks!